models/user.go

```diff
@@ -1226,11 +1229,13 @@ func ValidateCommitsWithEmails(oldCommits *list.List) *list.List {
 	for e != nil {
 		c := e.Value.(*git.Commit)
 
-		if v, ok := emails[c.Author.Email]; !ok {
-			u, _ = GetUserByEmail(c.Author.Email)
-			emails[c.Author.Email] = u
-		} else {
-			u = v
+		if c.Author != nil {
+			if v, ok := emails[c.Author.Email]; !ok {
+				u, _ = GetUserByEmail(c.Author.Email)
+				emails[c.Author.Email] = u
+			} else {
+				u = v
+			}
```

Comment:
If `c.Author` is nil, then `u` will keep its value from the previous iteration. Should we set `u` to nil if `c.Author` is nil?

Reply:
fixed